uaclient/mainwindow.py

```diff
@@ -205,6 +219,122 @@ def _update_subscription_model(self, node, value, timestamp):
             i += 1
 
 
+class GraphUI(object):
+
+    # use tango color schema (public domain)
+    colorCycle = ['#4e9a06ff', '#ce5c00ff', '#3465a4ff', '#75507bff', '#cc0000ff', '#edd400ff']
+    acceptedDatatypes = ['Decimal128', 'Double', 'Float', 'Integer', 'UInteger']
+
+    def __init__(self, window, uaclient):
+        self.window = window
+        self.uaclient = uaclient
+
+        # exit if the modules are not present
+        if not use_graph:
+            self.window.ui.graphLayout.addWidget(QLabel("pyqtgraph or numpy not installed"))
+            return
+        self._node_list = [] # holds the nodes to poll
+        self._channels = [] # holds the actual data
+        self._curves = [] # holds the curve objects
+        self.pw = pg.PlotWidget(name='Plot1')
+        self.pw.showGrid(x = True, y = True, alpha = 0.3)
+        self.legend = self.pw.addLegend()
+        self.window.ui.graphLayout.addWidget(self.pw)
+
+        self.window.ui.actionAddToGraph.triggered.connect(self._add_node_to_channel)
+        self.window.ui.actionRemoveFromGraph.triggered.connect(self._remove_node_from_channel)
+
+        # populate contextual menu
+        self.window.ui.treeView.addAction(self.window.ui.actionAddToGraph)
+        self.window.ui.treeView.addAction(self.window.ui.actionRemoveFromGraph)
+
+        # connect Apply button
+        self.window.ui.buttonApply.clicked.connect(self.restartTimer)
+        self.restartTimer()
+
+
+    def restartTimer(self):
+        # stop current timer, if it exists
+        if hasattr(self,'timer') and self.timer.isActive():
+            self.timer.stop()
+
+        # define the number of polls displayed in graph
+        self.N = self.window.ui.spinBoxNumberOfPoints.value()
+        self.ts = np.arange(self.N)
+        # define the poll intervall
+        self.intervall = self.window.ui.spinBoxIntervall.value()*1000
+
+        # overwrite current channel buffers with zeros of current length and add to curves again
+        for i,channel in enumerate(self._channels):
+            self._channels[i] = np.zeros(self.N)
+            self._curves[i].setData(self._channels[i])
+
+        # starting new timer
+        self.timer = QTimer()
+        self.timer.setInterval(self.intervall)
+        self.timer.timeout.connect(self.pushtoGraph)
+        self.timer.start()
+        
+    @trycatchslot
+    def _add_node_to_channel(self,node=None):
+        # TODO: check if node is of number type
```

Comment:
Yes, sorry. Forgot to remove this comment after implementation! Thanks for the hint.